api/src/main/java/com/codeforcommunity/rest/PostsRouter.java

```diff
@@ -130,4 +139,170 @@ private void handleGetCommentsForPost(RoutingContext ctx) {
       end(ctx.response(), 404, e.getMessage(), "text/plain");
     }
   }
+
+  /**
+   * Register the POST "/posts" route.
+   *
+   * @param router The Router to register the route with.
+   */
+  private void registerPostPostsRoute(Router router) {
+    Route route = router.post("/");
+    route.handler(this::handlePostPostsRoute);
+  }
+
+  /**
+   * Handle the POST "/posts" route.
+   *
+   * @param ctx The {@link RoutingContext} containing all relevant routing info.
+   */
+  private void handlePostPostsRoute(RoutingContext ctx) {
+    // Unmarshal the request body as a CreatePostRequest class.
+    CreatePostRequest createPostRequest = getJsonBodyAsClass(ctx, CreatePostRequest.class);
+    // Validate the DTO and ensure the provided info is valid.
+    if (!createPostRequest.validate()) {
+      end(ctx.response(), 400, "Create Post fields cannot be null.", "text/plain");
+      return;
+    }
+
+    // Create the post using the processor.
+    processor.createPost(createPostRequest);
+    // Return a success.
+    end(ctx.response(), 200, "Post created.", "text/plain");
```

Comment:
Oh I didn't realize a plain string is valid JSON. Also, I just ended up using 200 because it's simple. 201 is more correct though so I'll change to that.